src/expandableCalendar/index.tsx

```diff
@@ -255,29 +259,35 @@ const ExpandableCalendar = (props: ExpandableCalendarProps) => {
     }
   }, []);
 
-  useEffect(() => {
-    // date was changed from AgendaList, arrows or scroll
-    scrollToDate(date);
-  }, [date]);
-
   const handleScreenReaderStatus = (screenReaderEnabled: any) => {
     setScreenReaderEnabled(screenReaderEnabled);
   };
 
   /** Scroll */
+  const autoScroll = useRef(false);
+
+  const scrollToDate = throttle((date: string) => {
+    autoScroll.current = true;
 
-  const scrollToDate = (date: string) => {
     if (!horizontal) {
       calendarList?.current?.scrollToDay(date, 0, true);
     } else if (getYear(date) !== visibleYear.current || getMonth(date) !== visibleMonth.current) {
       // don't scroll if the month is already visible
       calendarList?.current?.scrollToMonth(date);
     }
-  };
+  }, 100, {leading: false, trailing: true});
+
+  /** Update Date */
 
-  const scrollPage = useCallback((next: boolean) => {
+  const newTempDate = useRef(date);
```

Comment:
This may lead to bugs. 
By creating a fork of the date value here we're losing the single source of truth principal. 

I see you update its value in `updatePage`, but technically the date value can be changed in many other places. 
This will probably lead to date issues in the future. 

I know we did something similar in `CalendarProvider`, which I'm also not a fan of, but at least the provider is where we maintain the original date value.
Once more sub components start to fork this value, we'll have a mess. 

If the reason you need this is for the debouncing the setDate, consider providing a `setDateDebounced` method from the context.
Components that need an immediate update will use `setDate`, others can use `setDateDebounced` 
In any case, it will be maintained by the provider

Reply:
Yes, I'm using the `newTempDate` ref to debounce the `setDate`. I need the date I'm passing `setDate` to be updated according to the new page without relaying on the date change by the`setDate` (context.date). It's a way to keep an updated date that I can increment/decrement (and scroll to it) even when I'm not really setting the new date to the context. I agree it can lead to bugs but I think we do use it often when we want to keep updated value without updating the state (and rerendering as a result). Even if I'll add a `debouncedSetDate` on the provider I will still need to keep the `newTempDate` ref locally as I need it for scrolling and moving the list items.

Reply:
But isn't it better to let the Provider handle it? 
Let's say another component will need to setDate with debounce, will we implement the logic again? 

Also, there's already a bug. I assume it's because you now maintain another copy of the date value. 

In the Expandable Calendar example. Scroll few weeks forward with gestures (no arrows), now scroll forward with the arrows. The date goes back, the dates are not synced.

Reply:
There is an issue with the WeekCalendar. I'll wait for the new implementation then...

Reply:
There is an issue with the WeekCalendar. I'll wait for the new implementation then...

Reply:
But if the issue is caused by the new date temp ref, why not fix that?

Reply:
I'm not sure it's related